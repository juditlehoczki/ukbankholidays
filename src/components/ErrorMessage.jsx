import React from "react";
import { Center, Button, Text } from "native-base";
import { MaterialIcons } from "@expo/vector-icons";

const ErrorMessage = ({ onPress, errorMessage }) => {
  return (
    <Center>
      <Text>{errorMessage}</Text>
      <Button
        size="sm"
        colorScheme="light"
        rounded="lg"
        rightIcon={<MaterialIcons name="refresh" size={30} color="white" />}
        onPress={onPress}
      >
        Try again
      </Button>
    </Center>
  );
};

export default ErrorMessage;
