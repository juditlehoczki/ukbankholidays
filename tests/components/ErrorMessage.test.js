import React from "react";
import { cleanup, fireEvent, render } from "react-native-testing-library";
import { NativeBaseProvider } from "native-base";

import ErrorMessage from "../../src/components/ErrorMessage";
import inset from "../inset";

describe("ErrorMessage Component", () => {
  let props;
  beforeEach(() => {
    props = {
      onPress: jest.fn(),
      errorMessage: "Test Error Message",
    };
  });
  afterEach(cleanup);

  test("renders correctly", () => {
    const component = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <ErrorMessage {...props} />
      </NativeBaseProvider>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
  test("button calls passed in onPress function", async () => {
    const component = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <ErrorMessage {...props} />
      </NativeBaseProvider>
    );
    const ButtonNode = await component.getByText("Try again");
    fireEvent.press(ButtonNode);
    expect(props.onPress).toHaveBeenCalled();
  });
});
