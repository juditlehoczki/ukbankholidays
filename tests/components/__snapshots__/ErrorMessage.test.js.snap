// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#1f2937",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      Test Error Message
    </Text>
    <View
      _text={
        Object {
          "color": "lightText",
        }
      }
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="light"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#78716c",
          "borderRadius": 8,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={Object {}}
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#FFFFFF",
                "fontFamily": undefined,
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "500",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textDecorationLine": undefined,
              }
            }
          >
            Try again
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "width": 8,
            }
          }
        />
        <Text />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
